core/src/main/scala/org/apache/spark/rdd/RDDOperationScope.scala

```diff
@@ -43,10 +43,9 @@ import org.apache.spark.SparkContext
 @JsonPropertyOrder(Array("id", "name", "parent"))
 private[spark] class RDDOperationScope(
     val name: String,
+    val id: String = RDDOperationScope.nextScopeId().toString,
```

Comment:
Super Nit: typically default params are ordered based on what is more likely to get explicitly specified. I think `parent` is more likely be specified by callers than the `id`, so should be ordered accordingly.